vortex-duckdb/src/exporter/dict.rs

```diff
@@ -70,6 +76,10 @@ impl<I: NativePType + AsPrimitive<u32>> ColumnExporter for DictExporter<I> {
         }
 
         vector.slice_to_dictionary(sel_vec, len);
+        // Use a unique id to each dictionary data array -- telling duckdb that the dict value vector
+        // is the same as reuse the hash in a join.
+        vector.set_dictionary_id(format!("{}-{}", self.cache_id, self.value_id));
+        vector.set_dictionary_len(self.values_len);
```

Comment:
What's this needed for?

Reply:
This is the duckdb API for reusing dict values (I missed this before)